### DISCUSSION TITLE
[Feature] Get job execution retries

### DISCUSSION THREAD

**Discussion by fooock** [jobrunr/jobrunr#545] [Ideas]
**Is your feature request related to a problem? Please describe.**
I want to see the number of retries for a given job

**Describe the solution you'd like**
Offer job metadata when `enqueue` a new job. Currently the job is automatically retried by using the filter but I want to know the number of retries for a particular job.

```java
jobScheduler.enqueue(() -> {
    var request = Request.<Event>builder()
          .url("http://localhost:8081")
          .payload(event)
          .build();
     eventClient.send(request);
});
```
I would like to have something like:

```java
jobScheduler.enqueue(() -> {
    var request = Request.<Event>builder()
          .url("http://localhost:8081")
          .payload(event)
          .build();
     eventClient.send(request);

}, jobMetadata -> {
   // immutable job metadata
});
```

**Comment by rdehuyss** [comment]
I don't understand what you want to achieve here to be honest. When you enqueue a new job, the number of retries is always 0.

So, giving access to the jobMetadata on creation will not add any benefit. You can already access the number of retries by using job filters.

The solution you propose will also not work in Kotlin as there you can only pass one lambda.

**Reply by patkujawa-wf** [reply]
Just to clarify one point - you can have many lambda args in kotlin, but the syntax sugar only works for the last lambda in the args list. You could have another method that takes two lambdas to keep everything backwards-compatible.

I don't understand the use case either, though 😆
